dataactbroker/handlers/aws/session.py

```diff
@@ -387,14 +307,15 @@ def newSession(uid,data,expiration) :
 
         Updates the exsiting session or creates a new one
         """
-        if(SessionTable.doesSessionExist(uid)) :
-            item = SessionTable.getTable().get_item(uid=uid)
-            item[SessionTable.DATA_FIELD] = data
-            item[SessionTable.DATE_FIELD] = toUnixTime(expiration)
-            item.save(overwrite=True)
-        else :
-            SessionTable.getTable().put_item(data={
-                SessionTable.KEY_NAME: uid,
-                SessionTable.DATA_FIELD: data,
-                SessionTable.DATE_FIELD: toUnixTime(expiration)
-            })
+        sess = GlobalDB.db().session
+        user_session = sess.query(SessionMap).filter_by(uid=uid).first()
```

Comment:
I'm not sure of the expected data in SessionMap, but if you want to make sure there's only one row per uid (maybe you don't!), you could use the SqlAlchemy `one_or_none()` function.